### DISCUSSION TITLE
Better Multiple Printer Support

### DISCUSSION THREAD

**Discussion by spikeon** [OrcaSlicer/OrcaSlicer#6357] [Ideas]
Hey guys, I have 5 VERY different printers that I use Orca Slicer to control.  I keep running into issues when building my projects and switching between printers; with settings or painting getting lost as different profiles are loaded and unloaded. 

For me, the dream solution would be to choose which printer is being used per-plate instead of per project, so I could lay out all the parts for my project and put the mechanical ABS parts on my X1C printer plates and the outer PLA housing on my A1, the small PLA buttons on my A1 Mini, and the large PLA stand and TPU foot on my Neptune 4 Max.  But I understand that this would require a major re-work of how the slicer functions at a fundamental level and isn't really a realistic suggestion.  Unless... ;) 

Moving on... My main suggestions are this: 

1. Make changing the printer part of the history.  You cannot "undo" a printer change.  Given how much changing a printer can break your project, this is just awful. 
2. It appears that when you switch to a printer it looks up the filaments that were selected the last time you worked with that printer.  The issue here is that if I was previously doing a single color print and I switch to that printer on a project where I have 10 colors, it removes all color allocations and breaks all of my painted objects and they need to be re-painted.  This gets even worse because you can't "undo". The fix for this would be to add filament slots with the same colors and materials if not enough filaments exist in the printer being loaded. I would be ok with this being optional with a prompt; such as "There are less filaments on the chosen printer, would you like to add more?"
3. When you switch printers or profiles the custom setting overrides that are the same as the chosen profile get lost.  This is an issue when I'm working on Printer A Profile 0.2mm and switch to Printer B.  It may select Profile 0.28 Ultra Draft and then I have to switch to Printer B Profile 0.2mm and in doing so I may lose project-custom settings that happened to match settings that were in Printer B 0.28 Ultra Draft.  This has happened to me more times than I can remember.   
Things that can help with this:
  a. When changing printers, ask us what profile we'd like to switch too instead of picking one for us. 
  b. Keep custom overrides marked as custom changes even if they match the profile that was switched too. 
  
I think that's all I can think of right now, but I'm sure there will be more notes as I continue using Orca.

**Comment by directme2dashow** [comment]
i was looking to make same comment / suggestion , i print projects and i send different parts to printers that handle different things better , like large or details or multicolor pieces,

**Comment by gg48gg** [comment]
I'd like to add..
If the intention is to be able to switch from the print settings in the main slicer window...
This is not working as intended when you have 2 of the same type of bambu printers. The only way to switch printers is to go to the devices tab, select the device, then switch back.

**Comment by DaKrampus** [comment]
I know this is an old post, but it is still relevant for me.. I think the main problem is that the device tab doesnt reflect the settings of the chosen printer.
example.. what slows me down in my workflow:
Printer 1 is on ip 1  (i enter the ip1 and port number in the field)
Printer 2 is on ip 2 (i enter the ip2 and port number in the field)
For both i also enter the IP with port 80 (for fluidd) in the device ui field.
It works like a charm for the first printer.
But as soon as i switch to printer 2.. and then go to the device tab it still is on printer 1
i cannot change it because there is no field (in the device tab) where i can change the ip.
As I have 6 printers installed.. it can get very confusing when i am working on a bigger project.
even after closing and restarting orca it stays on that device tab.. even after the printer is switched. probably has something to do with the browser cache that is not flushed when switching printer.
Sorry if i posted this in the wrong place.

**Comment by 3DNetherlands** [comment]
Since i've upgraded from 3 to 10 printers, i need something like this! The select printer per plate would be perfect, when you press the print button you can select or deselect the plates you want to send. Just like the timelapse, bed leveling & enable ams options.

**Comment by Blindsay04** [comment]
My take on it would be something like this, have multiple sections within a project for the different printers

<img width="2523" height="1415" alt="image" src="https://github.com/user-attachments/assets/2c679fc3-b08a-4e77-a9b9-3763cce76943" />

**Comment by Felix14-v2** [comment]
If someone is interested in a workaround: https://github.com/OrcaSlicer/OrcaSlicer/issues/9980#issuecomment-5486501226

Not so fancy like paid cloud routing services for print farms, but at least it's free

**Comment by dspears0809** [comment]
This would certainly be a welcome feature to manage printing for multiple printers from the same interface. I think PrusaSlicer 3.0 recently added this feature. Would love to see this in OrcaSlicer.
